tools/ansible/tasks/wsu/main.yaml

```diff
@@ -191,16 +206,36 @@
             src: "{{ tmp_dir.path }}/kubernetes/node/bin/kube-proxy.exe"
             dest: "{{ tmp_dir.path }}/kube-proxy.exe"
 
+        - name: Download checksum file for cni plugins
+          get_url:
+            url: "{{ cni_plugins_location }}.sha512"
+            dest: "{{ tmp_dir.path }}/cni_plugins_checksum_file"
+
+        # The checksum file content is in the format "<sha> <filename>". So to get SHA we need to extract only the <sha>
+        # from the file
+        - name: Get the SHA for cni plugins
+          shell: |
+            cat {{ tmp_dir.path }}/cni_plugins_checksum_file | awk -F ' ' '{ print $1 }'
+          register: cni_plugins_sha
+          failed_when: cni_plugins_sha.stdout == ""
+
+        # We match the SHA512 of the downloaded file with the one that is provided upstream to ensure integrity and
+        # verify that the download has completed successfully
+        - name: Download cni plugins
+          get_url:
+            url: "{{ cni_plugins_location }}"
+            dest: "{{ tmp_dir.path }}/cni_plugins.tgz"
+            checksum: "sha512:{{ cni_plugins_sha.stdout }}"
+
         - name: Create temporary cni directory
           file:
             path: "{{ tmp_dir.path }}/cni"
             state: directory
 
-        - name: Get cni plugins
+        - name: Extract cni plugins
```

Comment:
I think we will want to get rid of the .tgz. Leaving it results in the .tgz being copied to the Windows instance, even though its not used.

Reply:
Nice catch, @sebsoto

Reply:
Aren't we doing the same with the kube node binaries?

Reply:
If that is case then we need to fix that and not repeat that bad pattern.

Reply:
Created a playbook task 'Remove cni plugins tar' to address this